src/shared/base-filesystem.c

```diff
@@ -46,6 +46,9 @@ static const BaseFilesystem table[] = {
          * create the symlink as needed should it be missing. We prefer doing this consistently with Debian's
          * multiarch logic, but support Fedora-style multilib too.*/
 #if defined(__aarch64__)
+        { "lib64",    0, "usr/lib/aarch64-linux-gnu\0"
+                         "usr/lib64\0",                "ld-linux-aarch64.so.1" },
```

Comment:
Hmm, now I am puzzled. The Debian doc you linked suggests lib64 is not used there? but on fedora it is?

But either way ld-linux-aarch64.so.1 is not in lib64 but lin lib hence not the right thing to look for here. i.e. it appears there should not be any conditionalization on ld.so existance here.

Reply:
> but on fedora it is?

The directory is there. I included the listing in the commit message so if somebody wants to figure out why we added that, they'll know we didn't make it up ;)

The symlink is also there: `/lib64/ld-linux-aarch64.so.1 -> ld-2.32.so`.

Reply:
the point I was making is that I am not sure we need the lib64 symlink at all on this arch. I mean, in an ideal world the root fs would have no such symlinks at all. But unfortunately the ELF ABI of some archs declares that the ld.so dynamic loader is in /lib64/ hence if you want to be able to run any standards conforming dynamic ELF binary you have to have the lib64 symlink in place.

But, aarch64 is not one like that afaics, the official ld.so path for it is actually /lib/ld-linux-aarch64.so.1, not /lib64/ld-linux-aarch64.so.1, thus the necessity to have the /lib64 symlink in place is just not there.

For any other .so but the dynamic loader .so itself it's entirely sufficient that it appears in the default .so search path. The dynamic linker one is different however, it's path is hardcoded in the ELF header and part of the architecture ABI. Thus, for regular .so files we can just rely that whatever the shared libs are located below /lib the search logic will find it, only for the dynamic loader .so itself we need to make the symlink. Hence, I am questioning if we need the above at all...

Reply:
(but maybe I missing something because #14311 actually talks about /lib64/ specifically, hence I am a bit confused)

Reply:
I think it's better to create the symlink, if it is already present in "normal" installations. It might not matter for the linker, but people occasionally use paths, and a path with `/lib64/` would normally work.

Reply:
but that means we'd never have "legacy-free" systems that don't bother with /lib64/ at all, we'd always carry that cruft around...

I mean, we could certainly do that, but if that's the idea then looking for the right ld.so file in that dir only makes vague sense, because the 4th field in the struct is about that: find the right dir below /usr/ that has the ld.so in it, because we need to hook it up. But if we don't actually need to hook it up, then we don't have to insist in it (and most likely should link /lib64 to /lib probably)

Reply:
Oh, I checked the symlinks more, and it's actually the same situation as described in #14311. ` /lib/ld-linux-aarch64.so.1 -> ../lib64/ld-2.32.so`, so we **must** link `/lib64 → usr/lib64`. The patch was correct, but I'll extend the description a bit.

Reply:
hmm, ok, if fedora put their dynamic loader into lib64 even though the ABI defined lib for it, and then they fixed it up with a symlink, then I guess it indeed is the rght approach to symlink /lib64 to /usr/lib64 too, and also the check for the dynamic loader does indeed make sense.

Could you add a quick comment to the sources about this? Something like this:

```c
/* aarch64 ELF ABI actually says dynamic loader is in /lib/, but Fedora puts it in /lib64/ anyway and just symlinks /lib/ld-linux-aarch64.so.1 to ../lib64/ld-linux-aarch64.so.1. For this to work correctly /lib64/ must be symlinked to /usr/lib64/, hence do this in any case. */
```